packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -920,8 +920,19 @@ export class PipelineEditor extends React.Component<
 
   async handleOpenPipeline(): Promise<void> {
     this.widgetContext.ready.then(async () => {
-      let pipelineJson: any = this.widgetContext.model.toJSON();
-      if (pipelineJson == null) {
+      let pipelineJson: any = null;
+
+      try {
+        pipelineJson = this.widgetContext.model.toJSON();
+      } catch (error) {
+        if (error instanceof SyntaxError) {
```

Comment:
Is this a wide-enough net?  Could any other error could get us into the same data loss situation or is the fact that such errors would now be ignored prevent that from happening?

Reply:
That's a good point, I'll remove the condition